packages/web-fragments/src/gateway/pagesMiddleware.ts

```diff
@@ -63,12 +64,39 @@ export function getPagesMiddleware(
           element(element) {
             const scriptType = element.getAttribute('type');
             if(scriptType) {
-                element.setAttribute('data-script-type', scriptType);
+              element.setAttribute('data-script-type', scriptType);
             }
             element.setAttribute('type', 'inert');
           }
         });
-        const fragmentRes = scriptRewriter.transform(await fetch(upstreamUrl, { ...request }));
+
+        const fragmentReq: RequestInfo = { ...request, url: upstreamUrl };
+        let fragmentRes: Response;
+        let fragmentFailedResOrError: Response | unknown | null = null;
+        try {
+          fragmentRes = scriptRewriter.transform(await fetch(fragmentReq));
+          if (fragmentRes.status >= 400 && fragmentRes.status <= 500) {
```

Comment:
should the range here be <= 599? Otherwise we wouldn't handle e.g. 502 Bad Gateway errors

Reply:
yeah 🤦 sorry silly distraction mistake...